### PR TITLE
feat(linter): implement eslint rule `no-return-await`

### PR DESCRIPTION
**Pull request by vagusX**

### CODE REVIEW

**Comment by vagusX** [comment]
Eslint test cases are more than 100, and clippy warning this, @Boshen can we change this config to allow more test cases code?
image

File: crates/oxc_linter/src/rules/eslint/no_return_await.rs [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    /// https://eslint.org/docs/latest/rules/no-return-await
+    ///
+    /// ### Example
+    /// ```javascript
+    /// async function foo() {
+    ///   return await bar();
+    /// }
+    /// ```
+    NoReturnAwait,
+    correctness
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/no_return_await.rs:34]: This is not part of eslint recommended, but the extra `return await` seems useless to me 🤔 Let's keep this as correctness and see if there are false positives.

You can test rule alone with `-A all -D no-return-await` in your internal codebase and see if anything comes up?

File: crates/oxc_linter/src/rules/eslint/no_return_await.rs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        }
+    }
+    false
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/no_return_await.rs:141]: Try running `just new-rule no-return-await` and copy the tests over, so we can get rid of `\n` and `\t`

**Review by Boshen — CHANGES_REQUESTED** [review]

**Comment by Boshen** [comment]
> Eslint test cases are more than 100, and clippy warning this, @Boshen shall we change this config to allow more test cases code? image

Tell clippy to shut up with `allow`.

File: crates/oxc_linter/src/rules/eslint/no_return_await.rs (hunk 2 above)
Comment (vagusX) [reply crates/oxc_linter/src/rules/eslint/no_return_await.rs:145]: I found that we would lose the last 8 test cases yesterday.
image

Comment (vagusX) [reply crates/oxc_linter/src/rules/eslint/no_return_await.rs:146]: Updated, but the last 8 test cases are copied by hand

File: crates/oxc_linter/src/rules/eslint/no_return_await.rs (hunk 1 above)
Comment (vagusX) [reply crates/oxc_linter/src/rules/eslint/no_return_await.rs:34]: I will try it later

**Review by Boshen — APPROVED** [review]

**Comment by Boshen** [comment]
Nice work! Thank you!

**Pull request merged**